File: algorithms/src/algorithms/linkedlist.rs
pub struct LinkedList<T> {
    value: T,
    next: Option<Box<LinkedList<T>>>,
}

// impl<T> From<Vec<T>> for LinkedList<T> {
//     fn from(value: Vec<T>) -> Self {
//     }
// }

impl<T> LinkedList<T> {
    pub fn new(value: T) -> Self {
        Self { value, next: None }
    }
}

File: book/java/algorithms/algorithms/miscellaneous/LinkedList.java
package algorithms.miscellaneous;

import java.util.Optional;

public class LinkedList<T> {
    public final T value;
    public Optional<LinkedList> next;

    public LinkedList(T value, Optional<LinkedList<T>> next) {
        this.value = value;
        this.next = next;
    }
}
